tensorflow/core/util/mkl_util.h

```diff
@@ -633,7 +633,8 @@ class MklDnnShape {
   /// also be Blocked format.
   inline void SetTfLayout(size_t dims, const memory::dims& sizes,
                           memory::format format) {
-    CHECK_EQ(dims, sizes.size());
+    DCHECK(dims != sizes.size()) << "SetTfLayout: Number of dimensions is not"
```

Comment:
The condition in `DCHECK` must be true, otherwise it prints a message. So this should be `DCHECK(dims == sizes.size())`, or, better yet,
```suggestion
    DCHECK_EQ(dims, sizes.size()) << "SetTfLayout: Number of dimensions is not"
```
Also, `is not match` -> `does not match`.